src/tools/suggest-tests/src/static_suggestions.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{sug, Suggestion};
+
+// FIXME: perhaps this could use `std::lazy` when it is stablizied
+macro_rules! static_suggestions {
+    ($( $glob:expr => [ $( $suggestion:expr ),* ] ),*) => {
+        pub(crate) const STATIC_SUGGESTIONS: ::once_cell::unsync::Lazy<Vec<(&'static str, Vec<Suggestion>)>>
+            = ::once_cell::unsync::Lazy::new(|| vec![ $( ($glob, vec![ $($suggestion),* ]) ),*]);
+    }
+}
+
+static_suggestions! {
+    "*.md" => [
+        sug!("test", 0, ["linkchecker"])
+    ],
+
+    "library/std/src/lib.rs" => [
+        sug!("test", 0, ["library/std"])
+    ],
```

Comment:
Can you move this to a dynamic suggestion so it can be specific about which crate to test? Any of the following should work:
```
    ./x.py test library/alloc
    ./x.py test library/core
    ./x.py test library/panic_abort
    ./x.py test library/panic_unwind
    ./x.py test library/proc_macro
    ./x.py test library/std
    ./x.py test library/test
    ./x.py test library/unwind
```

hmm, now I'm wondering again why we chose to make this a separate tool instead of using reflection from the `should_run` functions ... do you happen to have a pointer to that discussion?

Reply:
I've addressed your first point.

In terms of the second point, I can't seem to find a good place to link to but the general point is that since suggestions are implemented in code (so they can be dynamic and we don't need `serde_json` or whatever), any time a suggestion is updated the code must rebuilt. Having to rebuild bootstrap for a minor suggestion change is too expensive so it was decided that bootstrap would just have a small unchanging shim which would parse a separate crate's output.

Reply:
Hmm, ok. I'm trying to remember why we decided to have "dynamic suggestions" in the first place instead of using reflection in builder.rs. But it doesn't need to block this PR; it's already quite useful and we can improve this in a follow-up.